packages/core/src/components/dialog/_dialog.scss

```diff
@@ -106,10 +106,17 @@ $dialog-header-padding: math.div($pt-grid-size, 2);
 
   .#{$ns}-icon-large,
   .#{$ns}-icon {
-    color: $pt-icon-color;
     flex: 0 0 auto;
     margin-left: -3px;
     margin-right: $dialog-padding * 0.5;
+
+    // only apply light color to icons that are not intent-specific
+    .#{$ns}-icon:not([class*="#{$ns}-intent"]) {
+      &.#{$ns}-icon-large,
+      &.#{$ns}-icon {
```

Comment:
Are you sure this is correct? I read this (simplified) as `.#{$ns}-icon .#{$ns}-icon.#{$ns}-icon:not([class*="#{$ns}-intent"])` (for the standard size case), which (1) nests the icon class in itself, and (2) duplicates the selector on the inner icon element. Unless there are specificity issues, I feel like `:not([class*="#{$ns}-intent"])` should be enough here (i.e. drop the part before the colon in R114, and drop R115-R116.

Reply:
Ah, yep, you're absolutely right! The additional icon class is duplicative. Removed here 3f710af985ab56268b3e61c6d3a0bb9e0baad5a3